tests/common/mod.rs

```diff
@@ -0,0 +1,211 @@
+        // Create an rfile in the tmp dir
+        let mut rfile = LineWriter::new(
+            File::create(format!("{}/{}", self.get_wd(), RFILE_NAME))
+                .expect("Could not write rfile"),
+        );
+        match fx_type {
+            FxType::Fastq => {
+                writeln!(
+                    rfile,
+                    "{}",
+                    &format!(
+                        "{}_1\t{}\t{}",
+                        prefix,
+                        self.file_string(&format!("{}_1_fwd.fastq.gz", prefix), TestDir::Input),
+                        self.file_string(&format!("{}_1_rev.fastq.gz", prefix), TestDir::Input),
+                    )
+                )
+                .unwrap();
+                writeln!(
+                    rfile,
+                    "{}",
+                    &format!(
+                        "{}_2\t{}\t{}",
+                        prefix,
+                        self.file_string(&format!("{}_2_fwd.fastq.gz", prefix), TestDir::Input),
+                        self.file_string(&format!("{}_2_rev.fastq.gz", prefix), TestDir::Input),
+                    )
+                )
+                .unwrap();
+            }
+            FxType::Fasta => {
+                writeln!(
+                    rfile,
+                    "{}",
+                    &format!(
+                        "{}_1\t{}",
+                        prefix,
+                        self.file_string(&format!("{}_1.fa", prefix), TestDir::Input),
+                    )
+                )
+                .unwrap();
+                writeln!(
+                    rfile,
+                    "{}",
+                    &format!(
+                        "{}_2\t{}",
+                        prefix,
+                        self.file_string(&format!("{}_2.fa", prefix), TestDir::Input),
+                    )
+                )
+                .unwrap();
+            }
+        };
+        RFILE_NAME
+    }
+
+    pub fn create_par_rfile(&self) -> &str {
+        let mut rfile = LineWriter::new(
+            File::create(format!("{}/{}", self.get_wd(), RFILE_NAME))
+                .expect("Could not write rfile"),
+        );
+
+        let paths = read_dir(self.file_string("par_test", TestDir::Input)).unwrap();
+        for path in paths {
+            let path_str = path.unwrap().path().display().to_string();
+            writeln!(rfile, "{path_str}\t{path_str}").unwrap();
+        }
+        RFILE_NAME
+    }
+}
+
+// Helper for multi sample fasta, where column order isn't conserved
+pub fn var_hash(aln_string: &[u8]) -> HashSet<Vec<char>> {
+    let fastq_align_out = String::from_utf8(aln_string.to_vec()).unwrap();
+    let mut sample_vecs: Vec<Vec<char>> = Vec::new();
+
+    // Read sample lines
+    for (idx, line) in fastq_align_out.lines().enumerate() {
+        if idx % 2 == 1 {
+            let line_vec: Vec<char> = line.chars().collect();
+            sample_vecs.push(line_vec);
+        }
+    }
```

Comment:
Some of these functions are not used so should be remove. `cargo clippy` should tell you which ones